src/lib/items/hooks/useItem.js

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { getProductById } from '../../../pages/api/Items';
+
+/**
+ *
+ * @param {number} productId
+ * @param {array} deps
+ * @returns [imgSrc, title, price, desc, tags, favoriteCount]
+ */
+export function useItem(productId = 0, deps = []) {
+  const [imgSrc, setImgSrc] = useState('');
+  const [title, setTitle] = useState('');
+  const [price, setPrice] = useState(0);
+  const [desc, setDesc] = useState('');
+  const [tags, setTags] = useState([]);
+  const [isFavorite, setIsFavorite] = useState(false);
+  const [favoriteCount, setFavoriteCount] = useState(0);
+
+  const fetchProduct = async () => {
+    try {
+      const result = await getProductById(productId);
+      setImgSrc(result.images[0]);
+      setTitle(result.name);
+      setPrice(result.price);
+      setDesc(result.description);
+      setTags(result.tags);
+      setIsFavorite(result.isFavorite);
+      setFavoriteCount(result.favoriteCount);
+    } catch (error) {
+      // 디폴트 값
+      setImgSrc(
+        'https://www.shutterstock.com/image-vector/default-ui-image-placeholder-wireframes-600nw-1037719192.jpg'
+      );
+      setTitle('-');
+      setPrice(0);
+      setDesc('-');
+      setTags([]);
+      setIsFavorite(false);
+      setFavoriteCount(0);
+    }
+  };
+
+  useEffect(() => {
+    fetchProduct();
+  }, deps);
+
+  return [imgSrc, title, price, desc, tags, isFavorite, favoriteCount];
```

Comment:
리액트 useState는 튜플을 반환하지만 , 개인적인 경험으로는 커스텀훅은 객체를 반환하는게 좀 더 편하다고 느꼈습니다. 

```suggestion
 return {
    item
  }
```